From c1b631119cd094009d6136c866de6e7b921611cf Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 14 Aug 2019 23:36:45 +0200
Subject: [PATCH 15/51] Also add com.samsung.sensor.physical_proximity (if
 available, it is more a true proximity sensor than touch proximity sensor)

---
 .../com/android/server/display/DisplayPowerController.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index f3cf0938e069..187c01669575 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -533,6 +533,13 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
 
         if (!DEBUG_PRETEND_PROXIMITY_SENSOR_ABSENT) {
             mProximitySensor = mSensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY);
+            if(mProximitySensor == null) {
+                List<Sensor> sensors = mSensorManager.getSensorList(Sensor.TYPE_ALL);
+                for(Sensor sensor: sensors) {
+                    if("com.samsung.sensor.physical_proximity".equals(sensor.getStringType()))
+                        mProximitySensor = sensor;
+                }
+            }
             if(mProximitySensor == null) {
                 List<Sensor> sensors = mSensorManager.getSensorList(Sensor.TYPE_ALL);
                 for(Sensor sensor: sensors) {
-- 
2.25.1

